modules/security-understanding-container-validation.adoc

```diff
@@ -0,0 +1,159 @@
+// Module included in the following assemblies:
+//
+// * security/container_security/security-understanding.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="security-understanding-container-validation_{context}"]
+= Container security and validation
+
+{product-title} is released as a set of container images that comprise the different functions of the software.
+These images must be validated in order to ensure that they have not been tampered with by an external party, and to ensure that they are the intended images provided by Red{nbsp}Hat.
```

Comment:
Maybe we can rephrase to highlight "customer admins can use ClusterImagePolicy for their own policy too!" with something like:

> {product-title} runs containers from containers images.
> These images may be validated in order to ensure that they have not been tampered with by an external party.

Then the following sections are:

* _Image integrity validation at pull time_, which can be [dropped][1] or tweaked.
* _Image authenticity validation_, introducing ClusterImagePolicy as a generic tool.
* _Trusting an {product-title} installation_, which gets into Red Hat images for the first time.
  * Unfortunately for continuity, we have to take a bit of a detour into bootstrapping/installer trust, which is complicated.
  * _Trusting a connected installation_, still talking about client/installer trust.
  * _Trusting a disconnected installation_, still talking about client/installer trust.

Hrm, and seems like we want an additional section like _Trusting an {product-title} post-install_, that points out the trusted installer will roll out a trusted release image with [the `openshift` ClusterImagePolicy][2] (for TechPreview clusters), and that that pulls in the ClusterImagePolicy security discussed under _Image authenticity validation_ for images in the `quay.io/openshift-release-dev/ocp-release` scope.  That prevents folks from running an `quay.io/openshift-release-dev/ocp-release` image that hasn't been signed by the trusted Red Hat key used to sign OCP release images, e.g. in the event that quay.io or a customer-configured mirror registry is compromised.

Or maybe that only matters when we start to talk about updates, and [we're puning on that for now][3], because really, you'd need a compromised mirror and an admin tricked into updating to a malicious digest or a by-tag release image reference.

[1]: https://github.com/openshift/openshift-docs/pull/81985#discussion_r1767389420
[2]: https://github.com/openshift/enhancements/blob/master/enhancements/security/openshift-image-policy.md#cluster-update-keys
[3]: https://github.com/openshift/openshift-docs/pull/81985#discussion_r1767085205

Reply:
Where I’m a little lost is that the goal, per the title _of the PR_ (not of the document), is to talk about OCP product images … which don’t currently use sigstore / ``ClusterImagePolicy` IIRC (just in a tech preview). So if that is the goal, why bring `ClusterImagePolicy` into the picture at all, _now_? It would certainly be relevant in a future release, but I don’t think we are there yet.

OTOH the title of the document (as opposed to the PR) talks about container “security and validation”, i.e. not restricted to core OCP, and in that case `ClusterImagePolicy` is certainly the most relevant part, and the OCP trust bootstrapping a side issue.

Reply:
OCP trust bootstrapping is more of an installer issue, and I'm fine with them leading there, and doc'ing the long-GA processes around that.

ClusterImagePolicy is more of a node issue, and I'm fine with the node folks leading on that.  There was some work associated with the 4.16 tech-preview of the type in #76613, #77203, #78396.

I'm here because 4.17 is releasing the tech-preview `openshift` ClusterImagePolicy roughly as laid out in [this enhancement][1].  There's a release note landed today via #81997, but it doesn't cover mechanics, and I'd expect customer security folks to have some questions about mechanics.  But maybe they don't?  I'm fine letting this float until there's some clarity around the kinds of security questions customers ask, and if anyone is particularly interested in if/how to manage installer trust or have customers create their own ClusterImagePolicies.

[1]: https://github.com/openshift/enhancements/blob/master/enhancements/security/openshift-image-policy.md